dds/DCPS/NetworkConfigModifier.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "NetworkConfigModifier.h"
+
+#ifdef OPENDDS_NETWORK_CONFIG_MODIFIER
+
+#include "ace/os_include/os_ifaddrs.h"
+
+#include <net/if.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+NetworkConfigModifier::NetworkConfigModifier()
+{
+}
+
+bool NetworkConfigModifier::open()
+{
+  if (DCPS_debug_level > 0) {
+    ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) NetworkConfigModifier::open() enumerating interfaces\n")));
+  }
+
+  ifaddrs* p_ifa = 0;
+  ifaddrs* p_if = 0;
+
+  ACE_INET_Addr address;
+
+  if (::getifaddrs(&p_ifa) != 0)
+    return false;
+
+  // Using logic from ACE::get_ip_interfaces_getifaddrs
+  // but need ifa_name which is not returned by it
+  int count = 0;
+
+  // Pull the address out of each INET interface.
+  for (p_if = p_ifa; p_if != 0; p_if = p_if->ifa_next)
+  {
+    if (p_if->ifa_addr == 0)
+      continue;
+
+    // Check to see if it's up.
+    if ((p_if->ifa_flags & IFF_UP) != IFF_UP)
+      continue;
+
+    if (p_if->ifa_addr->sa_family == AF_INET) {
+      struct sockaddr_in *addr =
+        reinterpret_cast<sockaddr_in *> (p_if->ifa_addr);
+
+      // Sometimes the kernel returns 0.0.0.0 as the interface
+      // address, skip those...
+      if (addr->sin_addr.s_addr != INADDR_ANY) {
+        address.set((u_short) 0,
+                     addr->sin_addr.s_addr,
+                     0);
+
+        NetworkInterface iface(count, p_if->ifa_name, p_if->ifa_flags & IFF_MULTICAST);
+        iface.addresses.insert(address);
+
+        NetworkConfigMonitor::add_interface(iface);
+        NetworkConfigMonitor::add_address(count, address);
+
+        ++count;
+      }
+    }
+  }
+
+  ::freeifaddrs (p_ifa);
+
+  return true;
+}
+
+bool NetworkConfigModifier::close()
+{
+  if (DCPS_debug_level > 0) {
+    ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) NetworkConfigModifier::close()\n")));
+  }
+
+  return true;
+}
+
+void NetworkConfigModifier::add_interface(const OPENDDS_STRING &name)
+{
+  NetworkInterface* p_nic = 0;
+
+  ifaddrs* p_ifa = 0;
+  ifaddrs* p_if = 0;
+
+  ACE_INET_Addr address;
+
+  if (::getifaddrs(&p_ifa) != 0) {
+      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: NetworkConfigModifier::add_interface %p"),
+        ACE_TEXT("getifaddrs error - ")));
```

Comment:
The indent is off here, this needs to be controlled by debug level, and there's no newline at the end of the message.